plan.go

```diff
@@ -0,0 +1,206 @@
+/*
+ * go-leia
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package leia
+
+import (
+	"errors"
+
+	"go.etcd.io/bbolt"
+)
+
+// queryPlan is the interface for all query plans
+type queryPlan interface {
+	// execute the plan call the DocWalker for each matching document
+	execute(walker DocWalker) error
+}
+
+// defaultQueryPlan contains elements common for each query plan
+type defaultQueryPlan struct {
+	collection *collection
+}
+
+// fullTableScanQueryPlan is a query plan which scans all documents
+type fullTableScanQueryPlan struct {
+	defaultQueryPlan
+	queryParts []QueryPart
```

Comment:
isn't it odd that the other plans have a `query`, while this one has `queryParts` which is a subset of `query`?

Reply:
indeed, moved query to base